docs/documentation/server_admin/topics/account.adoc

```diff
@@ -20,6 +20,10 @@
 .Account Console
 image:images/account-console-intro.png[Account Console]
 
+You can also ask for additional scopes when calling the account console URL by setting the `scope` parameter in this format: _server-root_{kc_realms_path}/{realm-name}/account?scope=phone.
+The scopes can only be requested (and eventually granted) when the user is authenticating to the account console. Once authenticated, the granted scopes won't change
+until re-authenticating.
```

Comment:
> One clarifying question here: When reading "re-authenticating" here, it can be read that the user needs to login and logout for the scopes to change. Looking at the code, I wonder what happens if I call the account URL with different scope parameters and are then redirected to the authentication flow.

Do you want me to update the docs to be more clear about what "re-authenticating" means?

If the user is already authenticated, the scopes from the URL are ignored and JS adapter is initialized with the scopes available from the existing client session, which are the same as when the user authenticated for the first time.

The reason for keeping the original scopes is that I don't think it makes much sense to allow users to change the client's original intent when redirecting the user to the account console.

> 
> Second case: I already have the account console open, and then press refresh in the browser. Which scopes will then be active - the default ones only, and would the previous added scope be forgotten?

I think the above applies here too.

Reply:
I tested this and it behaves as you describe. I still wonder if this is an expected and useful behavior: If I for some reason want to have a scope set for the account console and therefore redirect the user to a URL with the scope set, how would I know that the user is not already logged in?

Reply:
I tested this and it behaves as you describe. I still wonder if this is an expected and useful behavior: If I for some reason want to have a scope set for the account console and therefore redirect the user to a URL with the scope set, how would I know that the user is not already logged in? 

One possible attempt to always add the scopes that are provided in the URL would be to change the method in the AccountConsole.java as follows:

```
    public Response getMainPage(@PathParam("path") String path) throws IOException, FreeMarkerException {
        if (auth == null || session.getContext().getUri().getQueryParameters().getFirst(OIDCLoginProtocol.SCOPE_PARAM) != null) {
            return redirectToLogin(path);
        }

        return renderAccountConsole();
    }
```

Still, this is surprising as it would only add scopes, never remove them (at least when looking at the response of the `/token` endpoint - the `index.ftl` of the account console will always show the initial `scope`?

When debugging the behavior more, I found that when opening the account console as an unauthenticated user, I'm starting the auth flow twice: Once as a 302 redirect, and another time from the `keycloak.js`  adapter that as there first redirect doesn't come back with a known state and challenge. It seems that this was a recent addition in #31469 to prevent the Account console flashing up, still it makes this more messy to debug than it should be.

In the issue, I read 

> By doing this, users don't need to manually change the URL when at the login page.

I might need to understand more when and how this would be used to see what the least-surprising behavior would be.

Reply:
> I still wonder if this is an expected and useful behavior: If I for some reason want to have a scope set for the account console and therefore redirect the user to a URL with the scope set, how would I know that the user is not already logged in?

It is better than before as we have been discussing with @mposolda. In any case, to achieve what you want a client can still issue an authorization request (e.g.: `http://localhost:8180/realms/userprofile/protocol/openid-connect/auth?client_id=account-console&redirect_uri=http%3A%2F%2Flocalhost%3A8180%2Frealms%2Fuserprofile%2Faccount&response_type=code&code_challenge=QVOxeaZnYB9ktZy3RNou_XqE6_G0FKL9rLuF4S9UjE8&code_challenge_method=S256&scope=phone`) to update the scopes. Including removing previously granted scopes.

This is how it was working before, no changes here.

Reply:
I still see "you can add a scope to the URL as long as you are currently not logged in" as brittle and a surprising behavior - especially if you share the link somewhere on a webpage you wouldn't know if the user is already logged in. I'd rather have it "if you specify the scope int he URL, those scope will be set (no matter what the previous scope were)". 

We can have a 1:1 call on this, or you can proceed regardless if someone else like @mposolda approves it.

Reply:
> I still see "you can add a scope to the URL as long as you are currently not logged in" as brittle and a surprising behavior

I know what you are saying but that is not really the case. You can still ask for additional scopes if you reach out the authorization endpoint directly. Just like it works today.

However, the support for the `scope` parameter in the account console URL has a specific meaning which is to allow clients to start an account console session with a pre-defined set of scopes. You can still request different scopes once the user is authenticated by sending a request directly to the authorization endpoint, just like it works today.

Reply:
@ahus1 It depends now on https://github.com/keycloak/keycloak/pull/35358. The last commit is the change set we are discussing here.

Reply:
> I still see "you can add a scope to the URL as long as you are currently not logged in" as brittle and a surprising behavior

+1 for the point above. IMO it can be nice that if the URL `/realms/test/account?scope=phone` is opened in the browser, the scope `phone` is always included and it does not depend on whether the account-console client was already authenticated before in some other browser tab. At least that looks to me as most usable behaviour from the UX perspective (but not an UX expert) :-)

But if I understand correctly, it works this way due the changes in https://github.com/keycloak/keycloak/pull/35358 ? As if scopes are taken from the token rather than for clientSession, the re-authentication will add scope `phone` to the token, so it will use the `phone` scope from the token (even if there was clientSession already existing without scope `phone`). Can you confirm if it works this way?

Reply:
Yeah, it works fine.

But the behavior is not really because of #35358. Is more because of what I thought would be the best behavior. 

Looks like I'm wrong so that I will update this PR to allow changing scopes whenever passing the `scope` parameter to the account console URL.

Thanks.

Reply:
@ahus1 I've updated the PR to do what you suggested.

Reply:
Thank you for the change. I can now see that the scopes change as expected when I call the account with the scope parameter.